server/spec/multi_org_spec.rb

```diff
@@ -679,4 +679,74 @@
     expect(@cp.get_pool(pool['id']).quantity).to eq(1)
     expect(@user_client.list_pool_entitlements(pool['id']).size).to eq(1)
   end
+
+  it 'uses a host-restricted pool from the sharing org' do
```

Comment:
Just added one.  The scenario of a guest not causing a heal in a host is actually handled in two places now: CandlepinPoolManager.entitleByProductsForHosts just returns an empty collection if the guest and host are in different orgs **and** Entitler.bindByProducts won't even find the guest's host since we scope the host search to just the guest's organization.  The former check has been in the code 2013 and the latter check I added recently.  I say we leave them both just to keep our bases covered.